hypershift-operator/controllers/sharedingress/router.go

```diff
@@ -172,7 +190,8 @@ func buildHCPRouterContainerMain() func(*corev1.Container) {
 			"haproxy",
 		}
 		// 4.16.0-rc.3 haproxy router image.
-		c.Image = "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:06e8c4156888f50be0ac46bf21337ecda329befa0f9a6bb1df8bce078ae04aaa"
+		// c.Image = "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:06e8c4156888f50be0ac46bf21337ecda329befa0f9a6bb1df8bce078ae04aaa"
```

Comment:
Please drop commented code, include a TODO here and incorporate same feedback than in https://github.com/openshift/hypershift/pull/4330/files#r1671941572